unstructured/ingest/connector/kafka.py

```diff
@@ -0,0 +1,315 @@
+        """
+        for chunk in self.chunks(dict_list, BATCH_SIZE):
+            num_uploaded += self.upload_msg(chunk)  # noqa: E203
+
+        producer = self.kafka_producer
+        producer.flush()
+        logger.info(f"Uploaded {num_uploaded} documents to Kafka")
+
+    def write(self, docs: t.List[BaseIngestDoc]) -> None:
+        content_list: t.List[t.Dict[str, t.Any]] = []
+        for doc in docs:
+            local_path = doc._output_filename
+            with open(local_path) as json_file:
+                dict_content = json.load(json_file)
+                for content in dict_content:
+                    content_list.append(
+                        {
+                            "type": content["type"],
+                            "text": content["text"],
+                            "filename": content["metadata"]["filename"],
+                        }
+                    )
+        self.write_dict(dict_list=content_list)
+
+
+@dataclass
+class KafkaIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    """Class encapsulating fetching a message and writing processed results."""
+
+    connector_config: SimpleKafkaConfig
+    raw_content: str
+    raw_filename: str
+    registry_name: str = "kafka"
+
+    def _tmp_download_file(self):
+        topic_file = self.connector_config.topic + "-" + self.raw_filename + ".json"
+        return Path(self.read_config.download_dir) / topic_file
+
+    @property
+    def version(self) -> t.Optional[str]:
+        return None
+
+    @property
+    def source_url(self) -> t.Optional[str]:
+        return None
+
+    @property
+    def filename(self):
+        """The filename of the file created"""
+        return self._tmp_download_file()
+
+    def _create_full_tmp_dir_path(self):
+        self._tmp_download_file().parent.mkdir(parents=True, exist_ok=True)
+
+    @property
+    def _output_filename(self):
+        """Create filename document id combined with a hash of the query to uniquely identify
+        the output file."""
+        output_file = self.connector_config.topic + ".json"
+        return Path(self.processor_config.output_dir) / output_file
+
+    @SourceConnectionError.wrap
+    @BaseSingleIngestDoc.skip_if_file_exists
+    def get_file(self):
+        self._create_full_tmp_dir_path()
+
+        pdf_data = base64.b64decode(self.raw_content)
+
+        with open(self.filename, "wb") as file:
+            file.write(pdf_data)
+
+
+@dataclass
+class KafkaSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    """Source connector for Kafka.
+    Main job is to consume from a Kafka topic and create instances of
+    KakfaIngestDoc.
+
+    Note that messages have the format of:
+    <filename>: the name of the file (with correct file extension)
+    <content>: base64 encoded (whether was binary or not)
+    """
+
+    connector_config: SimpleKafkaConfig
+    _consumer: t.Optional["Consumer"] = None
+
+    def check_connection(self):
+        pass
```

Comment:
Can we do something similar here to the destination connector? We will start using check_connection soon.